lib/alertmanager_api/silence.py

```diff
@@ -0,0 +1,198 @@
+        :return: a new silence object, including the silence_id
+        :rtype: SilenceDetails
+        :raises MissingMandatoryParamError:
+            if any of the required attributes are missing
+            in the silence object:
+            instance_name
+            start time
+            end time
+            author
+            comment
+        :raises requests.exceptions.RequestException:
+            when the request to the AlertManager failed
+        """
+        if not silence.instance_name:
+            msg = "Missing silence instance name"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.start_time_dt:
+            msg = "Missing silence start time"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.end_time_dt:
+            msg = "Missing silence end time"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.author:
+            msg = "Missing silence author"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+        if not silence.comment:
+            msg = "Missing silence comment"
+            self.log.critical(msg)
+            raise MissingMandatoryParamError(msg)
+
+        payload = {
+            "matchers": [
+                {
+                    "name": "instance",
+                    "value": silence.instance_name,
+                    "isRegex": False,
+                }
+            ],
+            "startsAt": silence.start_time_str,
+            "endsAt": silence.end_time_str,
+            "createdBy": silence.author,
+            "comment": silence.comment,
+        }
+        try:
+            api_url = f"{self.api_endpoint}/api/v2/silences"
+            response = requests.post(
+                api_url,
+                auth=self.auth,
+                headers={"Accept": "application/json"},
+                json=payload,
+                timeout=10,
+            )
+            response.raise_for_status()
+            silence_id = response.json()["silenceID"]
+            silence.silence_id = silence_id
+            return silence
+        except requests.HTTPError as req_ex:
+            msg = f"Failed to create silence in Alertmanager: {req_ex} "
+            if req_ex.response is not None:
+                msg += (
+                    "Failed to create silence. "
+                    f"Response status code: {req_ex.response.status_code} "
+                    f"Response text: {req_ex.response.text}"
+                )
+            self.log.critical(msg)
+            raise req_ex
+        except Exception as req_ex:
+            # to capture any other Exception that is not HTTPError
+            msg = f"Failed to create silence in Alertmanager: {req_ex} "
+            self.log.critical(msg)
+            raise req_ex
+
+    def remove_silence(self, silence):
```

Comment:
why do we need to provide silence details in order to remove the silence - surely we only need to provide ID?